src/react/webgl-renderer.js

```diff
@@ -131,9 +131,52 @@ export default class WebGLRenderer extends React.Component {
     }
   }
 
+  // Gets current size of canvas in css (logical/window) coordinates
+  _getCSSSize(canvas) {
```

Comment:
So this is one of trickiest aspects of the canvas API. Namely that `width` and `height` control the size of the drawing buffer, not the DOM element, and are set independently from the style.width, height (which are read back as` clientWidth` and `clientHeight`). We can't keep reading width and height since they will be multiplied by DPR, we'll keep multiplying the drawing buffer size. If there is a reflow issue with `clientWidth` and `clientHeight` we could use the `props.width` and `props.height` here as they are intended to be in logical coordinates.